astropy/coordinates/tests/test_sky_coord.py

```diff
@@ -1843,3 +1843,26 @@ def test_passing_inconsistent_coordinates_and_units_raises_helpful_error(kwargs,
     # https://github.com/astropy/astropy/issues/10725
     with pytest.raises(ValueError, match=error_message):
         SkyCoord(**kwargs)
+
+
+@pytest.mark.skipif(not HAS_SCIPY, reason="Requires scipy.")
+def test_match_to_catalog_3d_and_sky():
+    # Test for issue #5857. See PR #11449
+    cfk5_default = SkyCoord([1, 2, 3, 4] * u.degree, [0, 0, 0, 0] * u.degree, distance=[1, 1, 1.5, 1] * u.kpc,
+                            frame='fk5')
+    cfk5_J1950 = SkyCoord([1, 2, 3, 4] * u.degree, [0, 0, 0, 0] * u.degree, distance=[1, 1, 1.5, 1] * u.kpc,
```

Comment:
Sorry for being a bit pedantic, but I think the code is clearer if this is just
```
cfk5_J1950 = cfk4_default..transform_to(FK5(equinox='J1950'))
```

Reply:
You mean `cfk5_J1950 = cfk5_default.transform_to(FK5(equinox='J1950'))` right?

Reply:
I'm pretty sure the answer is "yes". It's really just a style thing: It's the same code, but a little shorter and thus faster to read and understand for the next person to work on this. (And technically also a tiny bit faster, because you don't have to create the new object, but that's probably not very important here.)

Reply:
I think it was a typo in mhvk's comment, about fk4 so I wanted to clarify that.

Reply:
Yes, that was a typo!